### DISCUSSION TITLE
is it possible to get httpOnly cookies in getServerSideProps?

### DISCUSSION THREAD

**Discussion by michaelwindell26** [vercel/next.js#17677] [Help] [answered]
I want to be able to set a `httpOnly` cookie like so

```
res.setHeader('Set-Cookie', .....)
```

and then get that cookie in `getServerSideProps` function like so

```
req.cookies.myCookie
```

Seems as though you cannot get `httpOnly` cookies in `getServerSideProps` function. Is this even possible?

**Comment by jamesmosier** [comment, marked as answer]
You should be able to like this:

```js
export const getServerSideProps = async ctx => {
  // get the cookies
  const cookies = ctx.req.headers.cookie

  // set the cookies
  ctx.res.setHeader('Set-Cookie', 'foo=bar; HttpOnly');

  return { props: { } };
};
```

**Reply by michaelwindell26** [reply]
Thanks @jamesmosier this is what I was trying and banging my head against the wall.

Turns out it was a weird one for me, (not sure if its specific to my project), but I was trying to set a cookie and it would not work in a specific api route but worked with the exact same code in another route.... eg:

/api/account/login = cookie did not set
/api/account-login = cookie set successfully

**Reply by orloxx** [reply]
Hi @jamesmosier, is there a way to filter out the cookies that are HTTP-only from `ctx.req.headers.cookie`?

**Reply by jamesmosier** [reply]
I do not think there are any unique identifiers between HttpOnly cookies and non-HttpOnly. I could be wrong, but HttpOnly just means that client side JavaScript cannot access them.

**Comment by trangchongcheng** [comment]
I have same problem, anyone helps me?
https://github.com/vercel/next.js/discussions/26597

**Comment by dclipca** [comment]
How is it possible for a ```httpCookie``` to be accessed from ```getServerSideProps```? isn't it against the spec of a ```httpOnly``` cookie not being able to be accessed from client Javascript??

**Reply by orloxx** [reply]
@dclipca `getServerSideProps` it's server-side, that's why it is possible.

**Reply by nilsfast** [reply]
Would you mind explaining, how it is possible for a cookie that an API sets in the browser to be read through `getServerSideProps`? 
Does Next.js store the cookies somehow, I thought that was the browsers job?

**Reply by MoltenCoffee** [reply]
@nilsfast Cookies are sent back and forth between the client and the server, which makes it a way to keep state. The client (browser) stores them, and sends them to the server with each request it makes. The server can read them and send commands to change the cookies back to the client, but does not store them.

**Comment by dclipca** [comment]
For clarification, I am able to access all cookies, including ```httpOnly```.

**Comment by dell-betterteem** [comment]
Hi guys need help :)
access and refresh token is set but i can't access it through SSR, it worked on localhost but in hosted url it doesn't log

![image](https://user-images.githubusercontent.com/102798329/165070898-ac4f5a39-e9ae-4b19-b18c-15c5a9bcf27f.png)

express api is hosted on azure web app

**Reply by TylerOlthuizen** [reply]
did you get this working?

**Reply by naimulemon** [reply]
I am stuck at same issue

**Reply by 0tii** [reply]
same here, on localhost I can access the httpOnly cookies inside getServerSideProps, but as soon as I am deployed (then, in prod, my backend also sets `secure` on the cookies) I am unable to read them in getServerSideProps.

**Reply by gare-dev** [reply]
same problem, any one knows the problem?
